README.rst

```diff
@@ -58,12 +58,164 @@ Python:
+along with other channel specific information—range, terminal configuration, and custom scaling—that formats the data.
+Refer to the `NI-DAQmx Channel <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/chans.html>`_ for more information.
+
+Example code to create a task to read data from an analog input channel:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> with nidaqmx.Task() as task:
+  ...     ai_channel = task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     ai_channel.ai_min = -10.0
+  ...     ai_channel.ai_max = 10.0
+  ...     task.read()
+  ...
+  -0.14954069643238624
+
+Example code to create a task to read data from two analog input channels:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> with nidaqmx.Task() as task:
+  ...     ai_channel_1 = task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     ai_channel_2 = task.ai_channels.add_ai_voltage_chan("Dev1/ai1")
+  ...     ai_channel_1.ai_min = -5.0
+  ...     ai_channel_1.ai_max = 5.0
+  ...     ai_channel_2.ai_min = -10.0
+  ...     ai_channel_2.ai_max = 10.0
+  ...     task.read()
+  ...
+  [-0.07477034821619312, 0.8642841883602405]
+
+Timing
+------
+You can use software timing or hardware timing to control when a signal is generated. With hardware timing,
+a digital signal, such as a clock on your device, controls the rate of generation. With software timing,
+the rate at which the samples are generated is determined by the software and operating system instead of
+by the measurement device. A hardware clock can run much faster than a software loop. A hardware clock is
+also more accurate than a software loop.
+Refer to the `Timing, Hardware Versus Software <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/hardwresoftwretiming.html>`_ for more information.
+
+Example code to acquire finite amount of data using hardware timing:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> from nidaqmx.constants import AcquisitionType, READ_ALL_AVAILABLE
+  >>> with nidaqmx.Task() as task:
+  ...     task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     task.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=10)
+  ...     data = task.read(READ_ALL_AVAILABLE)
+  ...     print("Acquired data: [" + ", ".join(f"{value:f}" for value in data) + "]")
+  ...
+  AIChannel(name=Dev1/ai0)
+  Acquired data: [-0.149693, 2.869503, 4.520249, 4.704886, 2.875912, -0.006104, -2.895596, -4.493698, -4.515671, -2.776574]
+
+TDMS Logging
+------------
+Technical Data Management Streaming (TDMS) is a binary file format that allows for high-speed data logging.
+When you enable TDMS data logging, NI-DAQmx can stream data directly from the device buffer to the hard disk.
+Refer to the `TDMS Logging <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/datalogging.html>`_ for more information.
+
+Example code to acquire finite amount of data and log it to a TDMS file:
+
+.. code-block:: python
```

Comment:
How do you read a TDMS file in Python? Is there a Python TDMS library that works with DAQmx's raw TDMS format? It looks like npTDMS has some support but I have never used it.

Reply:
TDMS logging is part of the requirement under the documentation feature scope. I have created example to read the data from TDMS file with npTDMS in another PR - #594 .